### ISSUE TITLE
Runner doesn't sometimes download all artifact files (due to the "too many open files" error)

### ISSUE THREAD

**Issue by kengz** [dstackai/dstack#99]
**Steps to reproduce**

1. Create `environment.yml`

```
name: mnist-env

channels:
  - pytorch
  - conda-forge
dependencies:
  - autopep8=1.4.4
  - pip=22.2
  - python=3.10.4
  - pytorch=1.12.0
  - torchvision=0.13.0
  - pip:
      - dstack==0.0.7
      - hydra-core==1.2.0
      - pytorch-lightning==1.6.5
```

2. Create `.dstack/workflows.yaml`:

```
workflows:
  - name: mnist-env
    help: setup Conda env as artifact for reuse by mounting
    provider: bash
    commands:
      - conda env create --file environment.yml --prefix env
      - conda clean -tiy
      - conda list
    artifacts:
      - path: env
  - name: mnist-env-test
    provider: bash
    deps:
      - tag: mnist-env
    commands:
      - chmod +x /workflow/env/bin/*
      - /workflow/env/bin/python -c 'import torch; print(torch.__version__)'
```

3. Run `mnist-env` and assign the `mnist-env` tag to it.

4. Run `mnist-env-test` workflow

Expected output:
```
1.12.0
```

Actual output:
```
ModuleNotFoundError: No module named 'site'
```

**Comment by peterschmidt85** [comment]
I added the following commands to `mnist-env-test`:

```
- du -s /workflow/env
- find /workflow/env -type f | wc -l
```

And ran the workflow several times.

Every time it failed, the output was:

```
9524136 /workflow/env
24746
```

2 times out of 7, the workflow worked, and the output was:

```
9592080 /workflow/env
31044
```

The experiment shows the problem occurs only sometimes and the reason is that not all artifact files are downloaded.

**Comment by peterschmidt85** [comment]
After adding additional logging, see errors in the runner now:

```
root@ip-172-31-29-108:/home/ubuntu# cat /var/log/cloud-init-output.log | grep level=error
time=2022-09-25T12:13:18Z level=error msg=Create file err=open /root/.dstack/tmp/user_artifacts/little-elephant-1/env/lib/python3.10/site-packages/torch/utils/dlpack.py: too many open files run_name=little-elephant-1 job_id=little-elephant-1,tmp-check-conda,0 workflow=tmp-check-conda
time=2022-09-25T12:13:18Z level=error msg=Create file run_name=little-elephant-1 job_id=little-elephant-1,tmp-check-conda,0 workflow=tmp-check-conda err=open /root/.dstack/tmp/user_artifacts/little-elephant-1/env/lib/python3.10/site-packages/torch/utils/hipify/__init__.py: too many open files
time=2022-09-25T12:13:18Z level=error msg=Create file run_name=little-elephant-1 job_id=little-elephant-1,tmp-check-conda,0 workflow=tmp-check-conda err=open /root/.dstack/tmp/user_artifacts/little-elephant-1/env/lib/python3.10/site-packages/torch/utils/hipify/__pycache__/__init__.cpython-310.pyc: too many open files
time=2022-09-25T12:13:18Z level=error msg=Create file workflow=tmp-check-conda err=open /root/.dstack/tmp/user_artifacts/little-elephant-1/env/lib/python3.10/site-packages/torch/utils/file_baton.py: too many open files run_name=little-elephant-1 job_id=little-elephant-1,tmp-check-conda,0
time=2022-09-25T12:13:18Z level=error msg=Create file err=open /root/.dstack/tmp/user_artifacts/little-elephant-1/env/lib/python3.10/site-packages/torch/utils/hipify/__pycache__/cuda_to_hip_mappings.cpython-310.pyc: too many open files run_name=little-elephant-1 job_id=little-elephant-1,tmp-check-conda,0 workflow=tmp-check-conda
```

**Comment by peterschmidt85** [comment]
The fix is ready and will get to the `0.0.8` update.
